models/DataObject/ClassDefinition/PreviewGeneratorInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+
+namespace Pimcore\Model\DataObject\ClassDefinition;
+
+
+interface PreviewGeneratorInterface
+{
+
+    /**
+     * @param \Pimcore\Model\DataObject\Concrete $object
+     * @param array $params
+     * @return string
+     */
+    public function generatePreviewUrl(\Pimcore\Model\DataObject\Concrete $object, array $params): string;
+
+
+    /**
+     * @return array
```

Comment:
Maybe the PHPDoc should be a bit more specific on what the returned array should look like.